### PR TITLE
[JBPM-8585] Business Central doesn't update a ServerTemplate after restarting the kie-server

### PR DESCRIPTION
**Pull request by elguardian**

### CODE REVIEW

**Comment by elguardian** [comment]
this is ok to test

**Comment by elguardian** [comment]
this is ok to test

**Comment by elguardian** [comment]
this is ok to test.

File: kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/KieServerSetup.java [hunk 1]
```diff
@@ -58,4 +63,19 @@ public void setServerConfig(KieServerConfig serverConfig) {
...
+            messages = new HashSet<>();
+        }
+        return messages;
+    }
+
+    public void setMessages(Set<Message> messages) {
+        this.messages = messages;
+    }
+
+    public boolean isRegistered() {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/KieServerSetup.java]: this method name is bit misleading as it indicates some kind of registration and the only thing it does is searches for error messages. So maybe we should change the name of the method to reflect it better...?

Comment (elguardian) [reply kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/KieServerSetup.java]: @mswiderski change it to hasNoErrors

File: kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestKieServerControllerImpl.java [hunk 2]
```diff
@@ -58,11 +68,15 @@ public Response connectKieServer(@Context HttpHeaders headers,
         logger.debug("Server info {}", serverInfo);
         KieServerSetup serverSetup = connect(serverInfo);
 
-        logger.info("Server with id '{}' connected", id);
-        String response = marshal(contentType, serverSetup);
+        if (serverSetup.hasNoErrors()) {
+            logger.info("Server with id '{}' connected", id);
+        } else {
+            logger.info("Server with id '{}' failed to connect", id);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestKieServerControllerImpl.java]: Would it have sense to raise log severity to warning?

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/ConnectedKieServerHandler.java [hunk 3]
```diff
@@ -53,15 +53,18 @@ public ConnectedKieServerHandler(WebSocketSessionManager manager, Session sessio
...
 
-        logger.info("Server with id '{}' connected", serverId);
-        String response = WebSocketUtils.marshal(serverSetup);
-        
-        return response;
+        if (serverSetup.hasNoErrors()) {
+            manager.addSession(serverInfo, session);
+            logger.info("Server with id '{}' connected", serverId);
+        } else {
+            logger.info("Server with id '{}' failed to connect", serverId);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/ConnectedKieServerHandler.java]: Same as above.

File: kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestKieServerControllerImpl.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestKieServerControllerImpl.java]: make sense.

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/ConnectedKieServerHandler.java (hunk 3 above)
Comment (elguardian) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/ConnectedKieServerHandler.java]: make sense

**Comment by elguardian** [comment]
@kiegroup/gatekeepers  can anybody merge this ?

**Comment by cristianonicolai** [comment]
@elguardian can you take a look in the reported issues from Sonar?

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
@cristianonicolai I did check all but for some reason sonar does not show up here.

**Comment by sutaakar** [comment]
@elguardian It seems that last build failed for some reason, that is why Sonar analysis hasn't been triggered. The build results are unfortunately gone already.

**Comment by sutaakar** [comment]
Jenkins retest this

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by mareknovotny** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this

File: kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/KieServerSetup.java [hunk 4]
```diff
@@ -58,4 +63,20 @@ public void setServerConfig(KieServerConfig serverConfig) {
...
+        }
+        return messages;
+    }
+
+    public void setMessages(Set<Message> messages) {
+        this.messages = messages;
+    }
+
+    public boolean hasNoErrors() {
+        return !getMessages().stream().anyMatch(e -> e.getSeverity().equals(Severity.ERROR));
```
Comment (cristianonicolai) [inline kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/KieServerSetup.java]: From Sonar: Replace this negation and "anyMatch()" with "noneMatch()".

**Comment by elguardian** [comment]
@cristianonicolai can we proceed ?

**Pull request merged**